query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/QueryBuilderDsl.java

```diff
@@ -0,0 +1,742 @@
+   */
+  static Selector getFunction(
+      CqlIdentifier keyspaceId, CqlIdentifier functionId, Iterable<Selector> arguments) {
+    return new FunctionSelector(keyspaceId, functionId, arguments);
+  }
+
+  /** Var-arg equivalent of {@link #getFunction(CqlIdentifier, CqlIdentifier, Iterable)}. */
+  static Selector getFunction(
+      CqlIdentifier keyspaceId, CqlIdentifier functionId, Selector... arguments) {
+    return getFunction(keyspaceId, functionId, Arrays.asList(arguments));
+  }
+
+  /**
+   * Shortcut for {@link #getFunction(CqlIdentifier, CqlIdentifier, Iterable)}
+   * getFunction(CqlIdentifier.fromCql(functionName), arguments)}.
+   */
+  static Selector getFunction(
+      String keyspaceName, String functionName, Iterable<Selector> arguments) {
+    return getFunction(
+        CqlIdentifier.fromCql(keyspaceName), CqlIdentifier.fromCql(functionName), arguments);
+  }
+
+  /** Var-arg equivalent of {@link #getFunction(String, String, Iterable)}. */
+  static Selector getFunction(String keyspaceName, String functionName, Selector... arguments) {
+    return getFunction(keyspaceName, functionName, Arrays.asList(arguments));
+  }
+
+  /**
+   * Shortcut to select the result of the built-in {@code writetime} function, as in {@code SELECT
+   * writetime(c)}.
+   */
+  static Selector getWriteTime(CqlIdentifier columnId) {
+    return getFunction(WRITETIME, getColumn(columnId));
+  }
+
+  /**
+   * Shortcut for {@link #getWriteTime(CqlIdentifier)
+   * getWriteTime(CqlIdentifier.fromCql(columnName))}.
+   */
+  static Selector getWriteTime(String columnName) {
+    return getWriteTime(CqlIdentifier.fromCql(columnName));
+  }
+
+  /**
+   * Shortcut to select the result of the built-in {@code ttl} function, as in {@code SELECT
+   * ttl(c)}.
+   */
+  static Selector getTtl(CqlIdentifier columnId) {
+    return getFunction(TTL, getColumn(columnId));
+  }
+
+  /** Shortcut for {@link #getTtl(CqlIdentifier) getTtl(CqlIdentifier.fromCql(columnName))}. */
+  static Selector getTtl(String columnName) {
+    return getTtl(CqlIdentifier.fromCql(columnName));
+  }
+
+  /**
+   * Builds a relation testing a column.
+   *
+   * <p>This call must be chained with an operator, for example:
+   *
+   * <pre>{@code
+   * selectFrom("foo").getAll().where(isColumn("k").eq(bindMarker()));
+   * }</pre>
+   */
+  static ColumnRelationBuilder isColumn(CqlIdentifier id) {
+    return new DefaultColumnRelationBuilder(id);
+  }
+
+  /**
+   * Shortcut for {@link QueryBuilderDsl#isColumn(CqlIdentifier)
+   * isColumn(CqlIdentifier.fromCql(name))}
+   */
+  static ColumnRelationBuilder isColumn(String name) {
+    return isColumn(CqlIdentifier.fromCql(name));
+  }
+
+  /**
+   * Builds a relation testing a qualified component of collection (as of Cassandra 4, this can only
+   * be used for map values).
+   */
+  static ColumnComponentRelationBuilder isColumnComponent(CqlIdentifier columnId, Term index) {
```

Comment:
`column component` seems rather vague.  If it only applies to map, why not just call it `mapValue`?

Reply:
It could easily cover list and tuple elements as well: `WHERE l[0] = 1` (I'm surprised Cassandra did not implement lists at the same time as maps, feels pretty straightforward to me).

Reply:
On second thought, we can always add other methods (`isListElement`, `isTupleComponent`) that call this one if the need arises in the future. So I agree with `isMapValue`.